lfs/batcher.go

```diff
@@ -39,11 +42,20 @@ func (b *Batcher) Next() []Transferable {
 }
 
 // Exit stops all batching and allows Next() to return. Calling Add() after
-// calling Exit() will result in a panic.
+// calling Exit() will result in a panic, unless Reset() is called first.
 func (b *Batcher) Exit() {
+	atomic.StoreUint32(&b.exited, 1)
 	close(b.input)
 }
 
+// Reset allows an Exit()ed batcher to be re-started.
+func (b *Batcher) Reset() {
+	if atomic.CompareAndSwapUint32(&b.exited, 1, 0) {
+		b.input = make(chan Transferable, b.batchSize)
+		go b.acceptInput()
+	}
```

Comment:
On a closer look, this is just acting as a noop if `Reset` is called and the batcher has not been `Exit`ed. Operationally this is fine so I'm just going to document it as such for now.